src/cfmms.jl

```diff
@@ -190,3 +190,113 @@ function find_arb!(Δ::VT, Λ::VT, cfmm::GeometricMeanTwoCoin{T}, v::VT) where {
     Λ[2] = geom_arb_λ(v[2]/v[1], R[2], R[1], η, γ)
     return nothing
 end
+
+
+struct UniV3{T} <: CFMM{T}
+    @add_two_coin_fields
+    current_price :: T
+    current_tick_index :: Int #This is the index of the maximal tick with price lower than current_price in the ticks dictionary
+    ticks #This is the tick mapping sorted by price
+    function UniV3(R,γ,idx,current_price,current_tick_index,ticks)
+        γ_T, idx_uint, T = two_coin_check_cast(R, γ, idx)
+        return new{T}(
+            MVector{2,T}(R),
```

Comment:
Maybe reserves should be computed from the `tick_list`?

Reply:
I think in practice the tick list alone doesn't tell you the reserves because there are fees accumulated at tick as well

Reply:
Would we want to add those to the reserves in the tick list? I don't want duplicate info floating around where we have to remember to update both (or need to more formally define the CFMM inference to be implemented)

Reply:
ok so just checked on the status of how fees work and it turns out fees aren't immediately added to liquidity at tick, they need to be retrieved and then reinvested into the liquidity position before the changes to take effect. https://docs.uniswap.org/protocol/guides/providing-liquidity/collect-fees so I think the approach of just calculating the reserves from the tick mapping would work. But I'm not sure it is neccesary, I will implement it and then make it optional to construct this from the tick mapping if the reserves are not passed in.

Reply:
Sounds good.

I think we should make the design choice to ignore fee accumulation. The onus is on the user to update the reserves appropriately before calculating a route.